### ISSUE TITLE
Dropdown: Blank Group Label

### ISSUE THREAD

**Issue by marengga** [primefaces/primereact#7285]
### Describe the bug

Group label will be blank if OptionGroupLabel is set other than `label`.

After some debugging, I found that actual group label can be accessed inside `optionGroup` object.

```
<Dropdown 
  ...
  optionGroupLabel="name"
  ...
/>
```
Here's the issue, it'll be blank if we set other than "label"

```
<Dropdown 
  ...
  optionGroupLabel="optionGroup.name"
  ...
/>
```
It works, but is it intentional that it should be "optionGroup.[attributeName]" ?

### Reproducer

https://stackblitz.com/edit/stackblitz-starters-bgqusw?file=app%2Fpage.tsx

### System Information

```Shell
Binaries:
    Node: 21.2.0 - /usr/local/bin/node
    Yarn: 1.22.22 - /usr/local/bin/yarn
    npm: 10.2.4 - /usr/local/bin/npm
    Watchman: 2023.11.20.00 - /usr/local/bin/watchman
  npmPackages:
    primereact: ^10.8.3 => 10.8.3 
    react: ^18 => 18.2.0 
    tailwindcss: ^3.3.0 => 3.4.0
```

### Expected behavior

As documented, putting attribute name on `optionGroupLabel` should be enough

**Comment by melloware** [comment]
Agreed you should not need `optionGroup.[key]` to access it.  PR is welcome if you want to look into how to fix it.
